data/src/main/java/io/gnosis/data/repositories/ConnectivityRepository.kt

```diff
@@ -0,0 +1,45 @@
+package io.gnosis.data.repositories
+
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+
+class ConnectivityRepository(private val connectivityManager: ConnectivityManager) {
```

Comment:
This does not belong in the `:data` layer of the app. It is not a repository as it doesn't store any data and it is specific to Android. I think this belongs in the upper layer, in `:app`